torch/_inductor/mkldnn_lowerings.py

```diff
@@ -544,24 +547,79 @@ def qlinear_unary(
+                    if (
+                        isinstance(
+                            ir.InputsKernel.unwrap_storage_for_input(w_zp),
+                            ir.ComputedBuffer,
+                        )
+                        and w_zp.get_numel() == 0
+                    ) or (
+                        isinstance(
+                            ir.InputsKernel.unwrap_storage_for_input(w_zp),
+                            ir.ConstantBuffer,
+                        )
+                        and torch.equal(
+                            torch.zeros_like(V.graph.constants[w_zp.get_name()]),
+                            V.graph.constants[w_zp.get_name()],
+                        )
+                    ):
+                        return True
+                    else:
+                        return False
+
+                def _is_act_comptaible_with_onednn():
+                    # oneDNN doesn't currently have an optimized kernel for a vector activation zero-point.
+                    # So, if the activation zero point is a vector, we won't support it with oneDNN.
+                    # We can support the provided activation if:
+                    # 1. if x_zp is dummy (empty tensor) created to match an Inductor pattern
+                    # 2. If the activation is per-tensor quantized (x may even be uint8 in that case).
+                    nonlocal x_zp
+                    if (
+                        isinstance(
+                            ir.InputsKernel.unwrap_storage_for_input(x_zp),
+                            ir.ComputedBuffer,
+                        )
+                        and x_zp.get_numel() == 0
+                        and x.get_dtype() == torch.int8
+                    ) or (
+                        isinstance(
+                            ir.InputsKernel.unwrap_storage_for_input(x_zp),
+                            ir.ConstantBuffer,
+                        )
+                        and len(x_zp.get_layout().size) == 0
+                    ):
+                        return True
+                    else:
+                        return False
+
                 if (
-                    isinstance(
-                        ir.InputsKernel.unwrap_storage_for_input(x_zp),
-                        ir.ConstantBuffer,
-                    )
-                    and len(x_zp.get_layout().size) == 0  # Per tensor quant of act
-                    and isinstance(
-                        ir.InputsKernel.unwrap_storage_for_input(w_zp),
-                        ir.ConstantBuffer,
-                    )
-                    and torch.equal(
-                        torch.zeros_like(V.graph.constants[w_zp.get_name()]),
-                        V.graph.constants[w_zp.get_name()],
-                    )  # We only compensate MatrixB and assume B_zp is 0 to avoid the compensation of MatrixA
+                    _is_sym_quantized_wgt()
+                    and _is_act_comptaible_with_onednn()
```

Comment:
Seems lots of onednn related comment added in _is_sym_quantized_wgt and _is_act_comptaible_with_onednn. But these check using to decide using GEMM template or not. How did these onednn related comment connected to using the GEMM Template?

Reply:
The GEMM template can support more cases than oneDNN qlinear, but because of auto-tuning, it can only be used with  as a competitor to oneDNN qlinear. Thus, it double-checks that the arguments passed to oneDNN qlinear are compatible with oneDNN.

Reply:
> The GEMM template can support more cases than oneDNN qlinear

I am confused why here in the lowering phase, will we see cases than oneDNN qlinear doesn't support?

Reply:
No, but someone unfamiliar with various nunaces of oneDNN qlinear support might try supporting a new case.
If these checks wouldn't exist, then the code will crash later, and would be harder to debug.
This way, it'd be easier to debug.

Reply:
Still I think these checkers are used to decide using GEMM Template or not which involves the name of oneDNN or lots of oneDNN related comments seems stranger.

Reply:
I should probably move the oneDNN compatibility comments & code before `if use_max_autotune():`.

As for `_is_act_comptaible_with_onednn` (which, BTW, has a typo), it's a sanity check that the values passed to oneDNN qlinear would be supported by it (Otherwise, one would have to wait for oneDNN qlinear to crash to figure that out). 

Please advise if it's okay to move them instead of removing them. Thanks!

Reply:
Basically, those checks should have been asserts in `qlinear_unary`